### PR TITLE
Add CI and TagBot workflows

### PR DESCRIPTION
**Pull request by timholy**
travis.org is going away and travis.com will not be free.
Moreover, the 1.0 release didn't get a tag.
This switches to GitHub Actions for both activities.

simonster needs to set up a DOCUMENTER_KEY.
Instructions can be found at https://github.com/julia-actions/MassInstallAction.jl/blob/a7872f2837e68e37857d0a8ea5b57acb101d8c8b/demos/MaintenanceDec2020/maintenance.jl#L29-L41

### CODE REVIEW

**Comment by timholy** [comment]
Puzzling, but the workflow is running here: https://github.com/timholy/Reexport.jl/runs/1612316499

**Comment by fingolfin** [comment]
When you add a new workflow, it is only executed on your personal fork, not on the target repository -- this is for security reasons.

**Comment by fingolfin** [comment]
Someone should also update the badge in the README.

Oh, and I guess this PR closes #25 :-)

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+jobs:
+  test:
+    name: Julia ${{ matrix.version }} - ${{ matrix.os }} - ${{ matrix.arch }} - ${{ github.event_name }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        version:
+          - '1.0'
+          - '1'
```
Comment (ararslan) [inline .github/workflows/CI.yml:17]:
```suggestion
          - '1'
          - nightly
```
?

Comment (timholy) [reply .github/workflows/CI.yml:17]: That would be fine. This package is unlikely to have nightly-specific failures on a frequent basis. I'd approve that change.

Comment (timholy) [reply .github/workflows/CI.yml:17]: (FYI I can't merge PRs here.)

Comment (ararslan) [reply .github/workflows/CI.yml:17]: Nor can I give you that power, unfortunately. (I'm a collaborator but don't have full administrative privileges.)

**Comment by ararslan** [comment]
> simonster needs to set up a DOCUMENTER_KEY.

Seems unlikely to happen... :grimacing: This package doesn't use Documenter though, so is it really needed?

**Comment by timholy** [comment]
I'm no expert on these matters, but AFAIK TagBot needs some key for workflow-dispatch.

**Comment by ararslan** [comment]
Hmm, okay. I'm okay with continuing to make git tags manually, and since Pkg doesn't care about git tags, I think it's okay for now. If you agree then I'll just go ahead and merge.

**Comment by timholy** [comment]
Sounds good. Perhaps we could also consider getting @simonster to agree to transfer the repository, or to add you with admin privileges?

**Comment by ararslan** [comment]
That would definitely be ideal but no one has heard from him in quite some time, unfortunately.

**Pull request merged**